File: searches/interpolation_search/interpolation_search.py
def interpolation_search(arr, value):
	first = 0;
	last = len(arr) - 1
	while (arr[first] <= value and arr[last] >= value):
		mid = first + (((value - arr[first]) * (last - first)) // (arr[last] - arr[first]))
		if arr[mid] == value:
			return True
		elif arr[mid] > value:
			last = mid - 1
		else:
			first = mid + 1
		print(first, mid, last)
	return False

File: searches/interpolation_search/interpolation_search.c
int interpolation_search(int arr[], int n, int value){
	int first = 0;
	int last = n - 1;
	int mid = 0;
	while (arr[first] <= value && arr[last] >= value){
		mid = first + (((value - arr[first]) * (last - first)) / (arr[last] - arr[first]));
		if (arr[mid] == value){
			return 1;
		}
		else if (arr[mid] > value){
			last = mid - 1;
		}
		else{
			first = mid + 1;
		}
	}
	return 0;
}
